resource/api/auth.py

```diff
@@ -35,65 +90,49 @@ def get_openid_configuration(issuer_url: str) -> dict:
-        response = requests.post(
-            url=introspection_endpoint,
-            json={"token": token, "client_certificate": client_certificate},
-            auth=(conf.OAUTH_CLIENT_ID, conf.OAUTH_CLIENT_SECRET),
-            verify=False,
-        )
-        if response.status_code != 200:
-            log.error(f"introspection request failed: {response.text}")
-            raise AccessTokenValidatorError("Introspection request failed")
-
-    except requests.exceptions.RequestException as e:
-        log.error(f"introspection request failed: {e}")
-        raise AccessTokenValidatorError("Introspection request failed")
-    return response.json()
-
-
-def introspect(
-    client_certificate: str, token: str, x_fapi_interaction_id: Optional[str] = None
+def check_token(
+    client_certificate: str | None,
+    token: str,
+    x_fapi_interaction_id: Optional[str] = None,
 ) -> Tuple[dict, dict]:
     """
-    Introspection fails if:
-        1. Querying the token introspection endpoint fails
-        2. A token is returned with active: false
-        3. Scope is specified, and the required scope is not in the token scopes
-        4. No client certificate is presented
-    If introspection succeeds, return a dict suitable to use as headers
-    including Date and x-fapi-interaction-id, as well as the introspection response
+    Check token fails if:
+        1. Basic token checks (expiry, active, etc) fail
+        2. The token is not bound to the client certificate
+        3. The token is not correctly signed
+    If check succeeds, return a dict suitable to use as headers
+    including Date and x-fapi-interaction-id, as well as the check token result
     """
 
     # Deny access to non-MTLS connections
     if client_certificate is None:
         log.warning("no client cert presented")
-        raise AccessTokenNoCertificateError("No client certificate presented")
-    introspection_response = check_token(token, client_certificate)
-    log.debug(f"introspection response {introspection_response}")
-
-    # All valid introspection responses contain 'active', as the default behaviour
-    # for an invalid token is to create a simple JSON {'active':false} response
-    if (
-        "active" not in introspection_response
-        or introspection_response["active"] is not True
-    ):
-        raise AccessTokenInactiveError(
-            "Invalid introspection response, does not contain 'active' or is not True"
-        )
+        raise CertificateMissingError("No client certificate presented")
+    cert = parse_cert(client_certificate)
+    client_id = str(
+        cert.subject.get_attributes_for_oid(x509.NameOID.COMMON_NAME)[0].value
+    )
+    # get the jwks endpoints from well-known configuration
+    openid_config = get_openid_configuration(conf.ISSUER_URL)
+    jwks_uri = openid_config["jwks_uri"]
+    jwks_client = jwt.PyJWKClient(jwks_uri)
+    ssl._create_default_https_context = (
+        ssl._create_unverified_context
```

Comment:
Why is this necessary? We shouldn't have something like this in code which might be copied and pasted.